sdk/digitaltwins/Azure.DigitalTwins.Core/samples/DigitalTwinsClientSample/ComponentSamples.cs

```diff
@@ -36,17 +36,44 @@ public async Task RunSamplesAsync(DigitalTwinsClient client)
                 .Replace(SamplesConstants.ComponentId, componentModelId);
 
             // Then we create models
-            Response<IReadOnlyList<Models.ModelData>> createModelsResponse = await client
-                .CreateModelsAsync(new[] { newComponentModelPayload, newModelPayload });
-            Console.WriteLine($"Successfully created models Ids {componentModelId} and {modelId} with response {createModelsResponse.GetRawResponse().Status}.");
+            Response<IReadOnlyList<Models.ModelData>> createModelsResponse = await client.CreateModelsAsync(
+                new[]
+                {
+                    newComponentModelPayload,
+                    newModelPayload
+                });
+            Console.WriteLine($"Created models Ids {componentModelId} and {modelId} with response {createModelsResponse.GetRawResponse().Status}.");
 
             #region Snippet:DigitalTwinsSampleCreateBasicTwin
 
             // Create digital twin with component payload using the BasicDigitalTwin serialization helper
 
-            var basicDigitalTwin = new BasicDigitalTwin
+            var basicTwin = new BasicDigitalTwin
             {
-                Id = basicDtId
+                Id = basicDtId,
+                // model Id of digital twin
+                Metadata = { ModelId = modelId },
+                CustomProperties =
+                {
+                    // digital twin properties
+                    { "Prop1", "Value1" },
+                    { "Prop2", 987 },
+                    // component
+                    {
+                        "Component1",
+                        new ModelProperties
+                        {
+                            // model Id of component
+                            Metadata = { ModelId = componentModelId },
+                            // component properties
+                            CustomProperties =
+                            {
+                                { "ComponentProp1", "Component value 1" },
+                                { "ComponentProp2", 123 },
+                            },
+                        }
+                    },
+                },
```

Comment:
nice!

Reply:
It is nicer to do it all inline, I think. I was happy to learn you can initialize values in an object that is already instantiated (like the dictionaries or the Metadata classes).